clients/osv/src/funTest/kotlin/OsvServiceTest.kt

```diff
@@ -0,0 +1,113 @@
+
+private val VULNERABILITY_FOR_PACKAGE_BY_COMMIT_REQUEST = VulnerabilitiesForPackageRequest(
+    commit = "6879efc2c1596d11a6a6ad296f80063b558d5e0f"
+)
+private val VULNERABILITY_FOR_PACKAGE_BY_NAME_AND_VERSION = VulnerabilitiesForPackageRequest(
+    pkg = Package(
+        name = "jinja2",
+        ecosystem = "PyPI"
+    ),
+    version = "2.4.1"
+)
+private val VULNERABILITY_FOR_PACKAGE_BY_INVALID_COMMIT_REQUEST = VulnerabilitiesForPackageRequest(
+    commit = "6879efc2c1596d11a6a6ad296f80063b558d5e0c"
+)
+
+class OsvServiceTest : StringSpec({
+    "getVulnerabilitiesForPackage()  returns the expected vulnerability when queried by commit" {
+        val expectedResult = getAssetAsString("vulnerabilities-by-commit-expected-result.json")
+
+        val result = OsvService().getVulnerabilitiesForPackage(VULNERABILITY_FOR_PACKAGE_BY_COMMIT_REQUEST)
+
+        result should beSuccess()
+        JSON.encodeToString(result.getOrNull()) shouldEqualJson expectedResult
+    }
+
+    "getVulnerabilitiesForPackage() returns the expected vulnerability when queried by name and version" {
+        val expectedResult = getAssetAsString("vulnerabilities-by-name-and-version-expected-result.json")
+
+        val result = OsvService().getVulnerabilitiesForPackage(VULNERABILITY_FOR_PACKAGE_BY_NAME_AND_VERSION)
+
+        result should beSuccess()
+        JSON.encodeToString(result.getOrNull()) shouldEqualJson expectedResult
+    }
+
+    "getVulnerabilityIdsForPackages() return the vulnerability IDs for the given batch request" {
+        val requests = listOf(
+            VULNERABILITY_FOR_PACKAGE_BY_COMMIT_REQUEST,
+            VULNERABILITY_FOR_PACKAGE_BY_INVALID_COMMIT_REQUEST,
+            VULNERABILITY_FOR_PACKAGE_BY_NAME_AND_VERSION
+        )
+
+        val result = OsvService().getVulnerabilityIdsForPackages(requests)
+
+        result should beSuccess()
+        result.getOrThrow() shouldBe listOf(
+            listOf("OSV-2020-484"),
+            emptyList(),
+            listOf(
+                "PYSEC-2014-8",
+                "PYSEC-2014-82",
+                "PYSEC-2019-217",
+                "PYSEC-2019-220",
+                "PYSEC-2021-66"
+            )
+        )
+    }
+
+    "getVulnerabilityForId() returns the expected vulnerability for the given ID" {
+        val expectedResult = getAssetAsString("vulnerability-by-id-expected-result.json")
+
+        val result = OsvService().getVulnerabilityForId("GHSA-xvch-5gv4-984h")
+
+        result should beSuccess()
+        JSON.encodeToString(result.getOrThrow()) shouldEqualJson expectedResult
+    }
+
+    "getVulnerabilitiesForIds() return the vulnerabilities for the given IDs" {
+        val ids = setOf("GHSA-xvch-5gv4-984h", "PYSEC-2014-82")
+
+        val result = OsvService().getVulnerabilitiesForIds(ids)
+
+        result should beSuccess()
+        result.getOrThrow().map { it.id } shouldContainExactlyInAnyOrder ids
+    }
+})
+
+private val JSON = Json { prettyPrint = true }
```

Comment:
With the Kotest JSON matchers you don't need the pretty printing anymore and can straight use `OsvApiClient.JSON`.

Reply:
Same goes for the previous commit, BTW.